src/org/etools/j1939_84/controllers/part03/Part03Step16Controller.java

```diff
@@ -54,12 +59,43 @@
     @Override
     protected void run() throws Throwable {
         // 6.3.16.1.a. Turn the engine off.
+        incrementProgress("Part 3, Step 16 Turn Engine Off and keep the ignition key in the off position");
+        ensureKeyOffEngineOff();
+
         // 6.3.16.1.b. Confirm Fault A is still implanted according to the manufacturer’s instruction.
+        incrementProgress("Waiting for implant of Fault A according to the engine manufacturer's instruction");
+        if (!isDevEnv()) {
+            waitForFault("Part 6.3.16.1.b");
+        }
+
         // 6.3.16.1.c. Wait manufacturer’s recommended interval with the key in the off position.
+        incrementProgress("Waiting for manufacturer's recommended interval after the Fault A implant");
+        if (!isDevEnv()) {
+            waitForManufacturerInterval();
+        }
+
         // 6.3.16.1.d. Turn ignition key to the ON position.
         // 6.3.16.1.e. Observe MIL and Wait to Start Lamp in Instrument Cluster
         // 6.3.16.1.f. Start Engine after MIL and Wait to Start Lamp (if equipped) have extinguished.
         // 6.3.16.1.g. Wait as indicated by the engine manufacturer’s recommendations for Fault A.
+        if (!isDevEnv()) {
+            waitForEngineStart();
+        }
+    }
+
+    private void waitForManufacturerInterval(){
+        String message = "Please wait for the manufacturer's recommended interval with the key in off position" + NL;
+        message += "Press OK to continue the testing" + NL;
+        displayInstructionAndWait(message, "Part 6.3.16.1.c", WARNING);
```

Comment:
maybe should put the `isDevEnv()` inside the `displayInstructionAndWait()` method?

Reply:
Is part of the step 16 changes.

Reply:
Well, I guess it needs to be removed :)